src/main/java/seedu/address/model/order/Order.java

```diff
@@ -0,0 +1,93 @@
+package seedu.address.model.order;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Represents an Order made by a Person in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Order {
+
+    // Item fields
+    private final Item item;
+    private final Count count;
+
+    // Data fields
+    private final Date date;
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Order(Item item, Count count, Date date) {
+        requireAllNonNull(item, count, date);
+        this.item = item;
+        this.count = count;
+        this.date = date;
+    }
+
+    public Item getDescription() {
+        return item;
+    }
+
+    public Count getCount() {
+        return count;
+    }
+
+    public Date getDate() {
+        return date;
+    }
+
+    /**
+     * Returns true if both orders have the same item.
+     * This defines a weaker notion of equality between two orders.
+     */
+    public boolean isSameOrder(Order otherOrder) {
```

Comment:
Maybe the method can be renamed to `hasSameItem` instead? `isSameOrder` feels like it can't be true because every order is technically different even if the item and count are the same.

Reply:
I see, however I was following the method naming of the Person class where isSamePerson() also adhered to a weak notion of equality, of each person only having the same name. Would this be applicable?